pre_commit/parse_shebang.py

```diff
@@ -52,7 +52,7 @@ def _error(msg):
             _error('not found')
         return exe
     elif not os.access(orig, os.X_OK):
-        _error('not found')
+        _error('not executable')
```

Comment:
there should be a new case here, one for "does not exist" and one for "not executable"

the test you modified shows that this change is incorrect in fact

also you'd use `_error('is not executable')`

Reply:
Thanks for the review. So the check for existence would be:
```
elif not os.access(orig, os.F_OK)
```
and the check for executability would be:
```
elif not os.access(orig.os.F_OK)
```
Using the following mode codes:
```
os.F_OK: Tests existence of the path.
os.R_OK: Tests readability of the path.
os.W_OK: Tests writability of the path.
os.X_OK: Checks if path can be executed.
```

Reply:
`os.path.isfile` is the usual way to check for a file

Reply:
`os.path.isfile` will also catch a directory which is covered in `elif os.path.isdir(orig)`.

Reply:
you may need to reorder the checks then